tests/address_bar_and_search/test_suggestion_engine_selection.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from selenium.common.exceptions import StaleElementReferenceException
+from selenium.webdriver import Firefox, Keys
+from selenium.webdriver.support import expected_conditions as EC
+
+from modules.browser_object import Navigation
+
+sites = ["Google", "Amazon.com", "Bing", "DuckDuckGo", "eBay"]
+
+
+@pytest.mark.parametrize("site", sites)
+def test_search_suggestion_for_engine_selection(driver: Firefox, site: str):
+    """
+    C1365228 - Test to verify that when entering "@", a list of search engines is suggested, and upon selecting an
+    engine, the search is performed using the selected engine.
+    """
+    nav = Navigation(driver).open()
+    nav.type_in_awesome_bar("@")
+
+    # Use base selector to find all search engine suggestions
+    suggestion_list_items = nav.get_element("search-suggestion-list", multiple=True)
+
+    # Filter elements to find the one that matches the desired site
+    suggestion_list_item = None
+    for item in suggestion_list_items:
+        try:
```

Comment:
You might be able to avoid this if you add `"doNotCache"` to the element's groups.

Reply:
I did a second commit, with the "doNotCache" added to the element's groups, but intermittently i run into the same staleElement problem. Is there anything I could do on my current commit, or I should use a try-except block afterall?